src/symbol.cpp

```diff
@@ -65,7 +68,7 @@ void Symbol::importIndexes(std::queue<Symbol*>* list){
 
 IntervalSet* Symbol::exportIndex(){
     IntervalSet* tmp = this->range;
-    this->range = NULL;
+    this->range = NULL; // TODO: memory leak?
     delete this;
```

Comment:
I don't think this can lead to memory leak, because in the line above I'm keeping the reference "this->range" and after I return it. 
That line "this->range = NULL;" is to prevent the destructor (that has to be done) to delete the range while I need it.
Probably that way to exchange pointers is not good programming, maybe I have to think something better and more readable.